### ISSUE TITLE
Move ViewEnvironment into workflow-ui/core-common

### ISSUE THREAD

**Issue by zach-klippenstein** [square/workflow#1160]
It has no Android dependencies, it's entirely self-contained. Should probably review other code in this module as well to make sure it's in the right place.

**Comment by zach-klippenstein** [comment]
The `ViewEnvironment` factory function that takes a `ViewRegistry` can be left in the android module, since `ViewRegistry` hard codes a dependency on Android `View`.

Alternatively, `ViewRegistry` (and maybe `ViewEnvironment`?) could be made generic on the "view" type. I'm reluctant to do this because it feels backwards to make these concepts generic on a platform-specific view concept just to support a platform-independent UI framework that doesn't care about those types anyway. See https://github.com/square/workflow-kotlin-compose/issues/12 for a discussion of how to make the compose integration less dependent on the types from these artifacts.

**Comment by rjrjr** [comment]
Makes sense to me. It'd be fun to try to apply it to another view system, eg web — or CLI! It's clearly redundant in Compose, though, right?

**Comment by rjrjr** [comment]
And I agree that genericizing ViewRegistry violates YAGNI.

**Comment by zach-klippenstein** [comment]
Cool, I'll leave this in the 1.0 milestone.
